File: AppRemover/src/main/java/com/vs_unusedappremover/common/RunIfTimeElapsed.java
package com.vs_unusedappremover.common;

public abstract class RunIfTimeElapsed  implements Runnable {
	
	private long lastRunMillis;
	
	public void runIfElapsed(long interval) {
		long time = System.currentTimeMillis();
		if (time - lastRunMillis > interval) {
			lastRunMillis = time;
			run();
		}
	}
}

File: AppRemover/src/main/java/com/vs_unusedappremover/common/RunIfTimeElapsed.kt
package com.vs_unusedappremover.common

abstract class RunIfTimeElapsed : Runnable {

    private var lastRunMillis: Long = 0

    fun runIfElapsed(interval: Long) {
        val time = System.currentTimeMillis()
        if (time - lastRunMillis > interval) {
            lastRunMillis = time
            run()
        }
    }
}
